addons/account_payment/controllers/portal.py

```diff
@@ -1,15 +1,20 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo.http import request
-
+from odoo import http
 from odoo.addons.account.controllers import portal
+from odoo.addons.payment import utils as payment_utils
 from odoo.addons.payment.controllers.portal import PaymentPortal
+from odoo.http import request
 
 
 class PortalAccount(portal.PortalAccount, PaymentPortal):
 
     def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
+        # EXTENDS account
+
         values = super()._invoice_get_page_view_values(invoice, access_token, **kwargs)
+        # We want to show the dialog even when everything has been paid (with a custom message).
+        values['payment'] = kwargs.get('payment')
```

Comment:
Move inside `portal_page_values` below; it's weird to define keys of `values` at bit everywhere.

Reply:
I did, but I also need them when there's nothing to pay, so added it [here](https://github.com/odoo/odoo/pull/174812/commits/88b238de7dd603d20c863a8fc7373c24f800c041#diff-270b42179719fa0edd95e88da03c1f7c83c4ab8ecd5c70153070fb37e596bc61R19-R22) as well

Reply:
I didn't see that, but I think it's fine like you did :)